lib/IDE/Refactoring.cpp

```diff
@@ -218,13 +238,15 @@ class Renamer {
 
   bool labelRangeMatches(CharSourceRange Range, LabelRangeType RangeType, StringRef Expected) {
     if (Range.getByteLength()) {
-      CharSourceRange ExistingLabelRange =
-          Lexer::getCharSourceRangeFromSourceRange(SM, Range.getStart());
+      bool IsEscaped = false;
+      CharSourceRange ExistingLabelRange = getLeadingIdentifierRange(Range, IsEscaped);
       StringRef ExistingLabel = ExistingLabelRange.str();
+      bool IsSingleName = Range == ExistingLabelRange ||
+        (IsEscaped && Range.getByteLength() == ExistingLabel.size() + 2);
```

Comment:
Why don't we need to compare the name when `IsEscaped == true`?

Reply:
Neither needs to compare the content really. It's just trying to check there's only a single label in the given range (rather than separate arg and param labels) as in that case the external name is effectively '_'. `getLeadingIdentifierRange(`) guarantees `ExistingLabelRange` is a subrange of `Range` and the first or only label, so comparing their sizes is sufficient. I think I just wrote the non-escaped case that way because it was shorter than comparing the sizes. Would rewriting it like the below help?
```
bool IsSingleName = Range.getByteLength() == ExistingLabel.size() + (IsEscaped ? 2 : 0);
```

Reply:
Makes sense, thanks for clarifying!